AY2122S1-CS2103T-T17-3/tp
# Add field options

### Description




- Fixes #52 
- Fixes #33 with 9c4dcc8
- Introduces several field options to the`Person` attribute classes to allow developers to customise the constraints on the user input: 

Option | Description
--- | --- 
IS_BLANK_VALUE_ALLOWED | If set to true, the field is allowed to be blank (for string fields such as phone, name, etc) 
IS_NULL_VALUE_ALLOWED | If set to true, the field is allowed to be null (for int/Date fields such as LastMet, etc) 
DEFAULT_VALUE | The default value for the field. Set when user does not pass in the prefix on `Person` creation.

- I have set `Name` and `Email` to be the only required fields now for client creation. I have made the rest optional.

### Type of change


### How to test

Command | Behaviour 
--- | --- 
`add n/chris1 e/chris@gmail.com` | Should create the client without the need for other fields
`add n/chris2` | Will raise an incorrect format error since we are missing the email field here
`add n/chris3 e/chris@gmail.com p/83219123` | Should create the client since all the required fields are present 



### Checklist
